include/pybind11/stl_binders.h

```diff
@@ -0,0 +1,184 @@
+	void maybe_default_constructible(Class_ &) {}
+
+
+	template<typename U = T, typename std::enable_if< std::is_copy_constructible<U>{} >::type * = nullptr>
+	void maybe_copy_constructible(Class_ &cl) {
+		cl.def(pybind11::init< Vector const &>());
+	}
+	template<typename U = T, typename std::enable_if< !std::is_copy_constructible<U>{} >::type * = nullptr>
+	void vector_bind_maybe_copy_constructible(Class_ &) {}
+
+
+	template<typename U = T, typename std::enable_if< has_equal_operator<U>(0) >::type * = nullptr>
+	void maybe_has_equal_operator(Class_ &cl) {
+	    cl.def(pybind11::self == pybind11::self);
+	    cl.def(pybind11::self != pybind11::self);
+
+		cl.def("count", [](Vector const &v, T const & value) { return std::count(v.begin(), v.end(), value); }, "counts the elements that are equal to value");
+	}
+	template<typename U = T, typename std::enable_if< !has_equal_operator<U>(0) >::type * = nullptr>
+	void maybe_has_equal_operator(Class_ &) {}
+
+
+	template<typename U = T, typename std::enable_if< has_not_equal_operator<U>(0) >::type * = nullptr>
+	void maybe_has_not_equal_operator(Class_ &cl) {
+	    cl.def(pybind11::self != pybind11::self);
+	}
+	template<typename U = T, typename std::enable_if< !has_not_equal_operator<U>(0) >::type * = nullptr>
+	void maybe_has_not_equal_operator(Class_ &) {}
+
+
+	template<typename U = T, typename std::enable_if< has_insertion_operator<U>() >::type * = nullptr>
+	void maybe_has_insertion_operator(Class_ &cl, std::string name) {
+		cl.def("__repr__", [name](typename vector_binder<T>::Vector &v) {
+				std::ostringstream s;
+				s << name << '[';
+				for(uint i=0; i<v.size(); ++i) {
+					s << v[i];
+					if( i != v.size() -1 ) s << ", ";
+				}
+				s << ']';
+				return s.str();
+			});
+
+	}
+	template<typename U = T, typename std::enable_if< !has_insertion_operator<U>() >::type * = nullptr>
+	void maybe_has_insertion_operator(Class_ &, char const *) {}
+
+
+
+
+public:
+	vector_binder(pybind11::module &m, char const *name) {
+		Class_ cl(m, name);
+
+		cl.def(pybind11::init<>());
+
+		//maybe_constructible(cl);
+		maybe_default_constructible(cl);
+		maybe_copy_constructible(cl);
+
+		// Capacity
+		cl.def("empty",         &Vector::empty,         "checks whether the container is empty");
+		cl.def("size",          &Vector::size,          "returns the number of elements");
+		cl.def("max_size",      &Vector::max_size,      "returns the maximum possible number of elements");
+		cl.def("reserve",       &Vector::reserve,       "reserves storage");
+		cl.def("capacity",      &Vector::capacity,      "returns the number of elements that can be held in currently allocated storage");
+		cl.def("shrink_to_fit", &Vector::shrink_to_fit, "reduces memory usage by freeing unused memory");
+
+		// Modifiers
+		cl.def("clear",                                     &Vector::clear,     "clears the contents");
+		cl.def("push_back",    (void (Vector::*)(const T&)) &Vector::push_back, "adds an element to the end");
```

Comment:
I'd be greatly in favor of only implementing the Pythonic methods (this also what Boost.Python does). I suggest returning the ``class_`` instance in the ``vector_indexing_suite`` code so that the caller could still  add the C++ methods if they are strictly required in a specific project.